### PR TITLE
Add a `package.mode` config option with choices `preserve` and `bundle` 

### PR DESCRIPTION
**Pull request by mrm007**
This controls how Crackle generates output files.

- `bundle` rolls up output files into as few chunks as possible (default, current behaviour)
- `preserve` creates separate files for all modules using the original module names as file names. This is similar to Rollup's [`output.preserveModules`](https://rollupjs.org/configuration-options/#output-preservemodules), but allows more fine-grained control because we hook into [`output.manualChunks`](https://rollupjs.org/configuration-options/#output-manualchunks).

### Why?

We found some circular dependencies in Braid compiled code using [skott](https://github.com/antoine-coulon/skott). We don't have a lot of control over Rollup's chunking algorithm, so we need to resort to a 1:1 mapping of `src` to `dist` files (we don't have circular dependencies in source files).

#### Running skott on `src/index.ts`

```
➜ npx skott --showCircularDependencies --displayMode=graph --no-trackTypeOnlyDependencies src/index.ts

 Running skott from entrypoint: src/index.ts

 Processed 403 files (2827.077ms)

 ✓ no circular dependencies found (depth=Infinity)
```

#### With `@crackle/cli@0.14.0`

```
➜ npx skott --showCircularDependencies --displayMode=graph dist/index.mjs

 Running skott from entrypoint: dist/index.mjs

 Processed 141 files (370.166ms)

 ✖ (37) circular dependencies found
```

#### With snapshot `0.0.0-preserve-modules-20240206222819`

```
npx skott --showCircularDependencies --displayMode=graph dist/index.mjs

 Running skott from entrypoint: dist/index.mjs

 Processed 397 files (590.499ms)

 ✓ no circular dependencies found (depth=Infinity)
```

### CODE REVIEW

File: tests/__snapshots__/package/with-vocab/dist/index.ts.snap [hunk 1]
```diff
@@ -3,8 +3,8 @@
 const jsxRuntime = require("react/jsx-runtime");
 const React = require("react");
 const client$1 = require("react-dom/client");
-const _vocab_index_cjs = require("./.vocab/index.cjs");
-const client_vocab_index_cjs = require("./client.vocab/index.cjs");
+const _vocab_index_ts = require("./.vocab/index.cjs");
```
Comment (askoufis) [inline tests/__snapshots__/package/with-vocab/dist/index.ts.snap:6]: Unsure what caused this change. It also reads a bit weird, but it's bundled code so it doesn't really matter.

**Review by askoufis — APPROVED** [review]
Needs some docs in the README, but otherwise LGTM.

**Comment by askoufis** [comment]
https://github.com/seek-oss/crackle/actions/runs/7807161298/job/21295006643?pr=167#step:6:73 😢

**Comment by mrm007** [comment]
kick the can

File: tests/__snapshots__/package/with-vocab/dist/index.ts.snap (hunk 1 above)
Comment (mrm007) [reply tests/__snapshots__/package/with-vocab/dist/index.ts.snap]: I know why: [`3f52861` (#167)](https://github.com/seek-oss/crackle/pull/167/commits/3f52861e7f0c7f75b35483e3db5f26f85a109d6c)

**Comment by mrm007** [comment]
I'll update the README for this and #168 in a follow-up PR

**Pull request merged**